services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/AuthorizationManagerImpl.java

```diff
@@ -196,13 +199,14 @@ public AuthorizationStatus canAccess(UserInfo userInfo, String objectId, ObjectT
 				}
 			}
 			case DATA_ACCESS_REQUEST:
+				if (accessType == ACCESS_TYPE.DOWNLOAD) {
+					return dataAccessAuthorizationManager.canDownloadRequestFiles(userInfo, objectId);
+				} else {
+					return AuthorizationStatus.accessDenied("Unexpected access type "+accessType);
+				}
 			case DATA_ACCESS_SUBMISSION: {
 				if (accessType==ACCESS_TYPE.DOWNLOAD) {
-					if (isACTTeamMemberOrAdmin(userInfo)) {
-						return AuthorizationStatus.authorized();
-					} else {
-						return AuthorizationStatus.accessDenied("Download not allowed");
-					}
+					return dataAccessAuthorizationManager.canDownloadSubmissionFiles(userInfo, objectId);

```

Comment:
dataAccessAuthorizationManager.canAccess(userInfo, accessType, objectType, objectId);

We would like to refactor this code to be based on a factory using the object type.  Then there would be zero logic in this class.  This is work we can do in the future maybe?